dali/base/dautils.hpp

```diff
@@ -577,4 +577,28 @@ inline unsigned calcStripeNumber(unsigned partNum, const char *lfnName, unsigned
 interface INamedGroupStore;
 extern da_decl void remapGroupsToDafilesrv(IPropertyTree *file, INamedGroupStore *resolver);
 
+#ifdef NULL_DALIUSER_STACKTRACE
+static time_t lastNullUserLogEntry = (time_t)0;
+static CriticalSection nullUserLogCS;
+#define LOGNULLUSER(userDesc) \
```

Comment:
This would cleaner as an exported function when NULL_DALIUSER_STACKTRACE is defined, with the statics in the cpp (otherwise each module that includes dautils.hpp will have a separate set of statics): 
```
#ifdef NULL_DALIUSER_STACKTRACE
extern da_decl void logNullUser(IUserDescriptor *userDesc);
#else
inline void logNullUser(IUserDescriptor *userDesc) { }
#endif
```